src/RepeatingStructures/HexagonalLattice.jl

```diff
@@ -0,0 +1,129 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+abstract type HexagonalLattice{N,T} <: Basis{N,T} end
+
+hexe₁(::Type{T}=Float64) where{T<:Real} = SVector{2,T}(T(1.5),T(.5)*sqrt(T(3)))
+hexe₂(::Type{T}=Float64) where{T<:Real} = SVector{2,T}(T(1.5),T(-.5)*(sqrt(T(3))))
+
+struct HexBasis1{N,T} <: HexagonalLattice{N,T}
+    HexBasis1(::Type{T} = Float64) where{T<:Real} = new{2,T}()
+end
+export HexBasis1
+
+basis(a::HexBasis1{2,T}) where{T} = SVector{2,SVector{2,T}}(hexe₁(T),hexe₂(T))
+
+# coordinate offsets to move up, down, etc., numsteps hex cells in a hex lattice defined in HexBasis1
+hexup(numsteps = 1) = numsteps .* (1,-1)
+hexdown(numsteps = 1) = numsteps .* (-1,1)
+hexupright(numsteps = 1) = numsteps .* (1,0)
+hexdownleft(numsteps = 1) = numsteps .* (-1,0)
+hexdownright(numsteps = 1) = numsteps .* (0,1)
+hexupleft(numsteps = 1) = numsteps .* (0,-1)
+
+""" (i,j) offsets of a ring represented in the HexBasis1 lattice basis. The hexagonal grid is assumed to start from the bottom center hex cell. To generate larger rings repeat this pattern
+ ring 1 = (1, 0)      (1, -1)       (0, -1)       (-1, 0)       (-1, 1)       (0, 1)
+ ring 2 = (1, 0)(1, 0)(1, -1)(1, -1)(0, -1)(0, -1)(-1, 0)(-1, 0)(-1, 1)(-1, 1)(0, 1)(0, 1)
+"""
+const hexcycle = SVector{6,NTuple{2,Int64}}(hexupright(),hexup(),hexupleft(),hexdownleft(),hexdown(),hexdownright())
+
+
+"""returns (i,j) offsets of ring n defined in the HexBasis1 lattice basis."""
+ringoffsets(n::Int64) = ringoffsets(Val{n})
+function ringoffsets(::Type{Val{N}}) where N
+    temp = MVector{N*6,NTuple{2,Int64}}(undef)
+
+    for i in 1:6
+        for j in 1:N
+            temp[(i-1)*N + j] = hexcycle[i]
+        end
+    end
+    
+    return SVector{N*6,NTuple{2,Int64}}(temp)
+end
+export ringoffsets
+
+"""Returns ```centerpoint``` plus all its neighbors in a ring of size n. This is a hexagonal region of the lattice n times as large as the unit hexagon cell.
+
+Example:
+
+```
+Vis.@wrapluxor Vis.drawhexcells(50, hexregion((0,0),2))
+```
+"""
+function hexregion(centerpoint::Tuple{Int64,Int64}, n::Int64) 
+    f(i) = i==0 ? () : (neighbors(HexBasis1,centerpoint,i)...,f(i-1)...)
+    return (centerpoint,f(n)...)
+end
+export hexregion
+
+"""Returns all hex cells contained in the rings up to size n centered around centerpoint. Does not include centerpoint. Use hexregion for that."""
+function neighbors(::Type{HexBasis1}, centerpoint::Tuple{Int64,Int64},n) where{T}
```

Comment:
Seems like this function returns the list of neighbors at exactly n distance from the centerpoint, and not "up to". I think that is the region function.

Reply:
You are correct.